### PR TITLE
!str #17355: Don't use case classes for settings

### PR DESCRIPTION
**Pull request by drewhk**

### CODE REVIEW

**Comment by drewhk** [comment]
refs https://github.com/akka/akka/issues/17355

**Comment by drewhk** [comment]
Shoud we cover attributes? Also, should I check Http as well? I think yes, but feedback is welcome. /cc @jrudolph @sirthias

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSpec.scala [hunk 1]
```diff
@@ -529,7 +529,7 @@ class FlowSpec extends AkkaSpec(ConfigFactory.parseString("akka.actor.debug.rece
 
   "A broken Flow" must {
     "cancel upstream and call onError on current and future downstream subscribers if an internal error occurs" in {
-      new ChainSetup(faultyFlow, settings.copy(initialInputBufferSize = 1), toFanoutPublisher(initialBufferSize = 1, maximumBufferSize = 16)) {
+      new ChainSetup(faultyFlow, settings.withInputBuffer(initialSize = 1, maxSize = 1), toFanoutPublisher(initialBufferSize = 1, maximumBufferSize = 16)) {
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSpec.scala]: perhaps add a `withInputBuffer(size: Int)`?

**Comment by patriknw** [comment]
LGTM, I don't know about the attributes. Everything case class will be a pain to evolve (at least after experimental phase). The attributes typically carry one or a few values and we can always add a new one (naming is the difficult part).

**Pull request merged**